AmethystAPI/src/amethyst/runtime/mod_info/ModInfo.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include "Json.hpp"
+#include "amethyst/Log.h"
+#include <optional>
+#include <string>
+using json = nlohmann::json;
+
+namespace Amethyst {
```

Comment:
A few things:
- This should all be in the Mod class, we don't need a separate class we just need to move Mod into the API rather than specific to runtime.
- "_" filepath should be using camelCase for folder names & PascalCase for class name